core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -470,14 +470,14 @@ extract_vector_length(iType, Args...) {
 
 }  // namespace Impl
 
-Impl::PerTeamValue PerTeam(const int& arg);
-Impl::PerThreadValue PerThread(const int& arg);
+Impl::PerTeamValue PerTeam(const size_t& arg);
+Impl::PerThreadValue PerThread(const size_t& arg);
```

Comment:
Should we go ahead and fix that by taking the size argument by value?

Reply:
I would do it in another pull request throughout the whole repository.

Reply:
To be clear I am not encouraging such a PR.  I don't even want to think about the amount of noise it will generate.